src/coreclr/src/jit/morph.cpp

```diff
@@ -8435,6 +8435,10 @@ void Compiler::fgMorphTailCallViaJitHelper(GenTreeCall* call)
             thisPtr = objp;
         }
 
+        // TODO-Cleanup: we leave it as a virtual stub call to
```

Comment:
I made a try to clear `GTF_CALL_VIRT_KIND_MASK ` here but it caused failures because logic in `LowerCall` depends on this flag to be set in some cases, so I reverted that change.

It would be nice to clear it and fix the failures but it is out of the scope of this bug fix for CoreRT.